android/src/main/java/net/no_mad/tts/TextToSpeechModule.java

```diff
@@ -580,43 +576,11 @@ else if(ready != Boolean.TRUE) {
     }
 
     @SuppressWarnings("deprecation")
-    private int speak(String utterance, String utteranceId, ReadableMap inputParams) {
-        String audioStreamTypeString = inputParams.hasKey("KEY_PARAM_STREAM") ? inputParams.getString("KEY_PARAM_STREAM") : "";
-        float volume = inputParams.hasKey("KEY_PARAM_VOLUME") ? (float) inputParams.getDouble("KEY_PARAM_VOLUME") : 1.0f;
-        float pan = inputParams.hasKey("KEY_PARAM_PAN") ? (float) inputParams.getDouble("KEY_PARAM_PAN") : 0.0f;
-
-        switch(audioStreamTypeString) {
-            /*
-            // This has been added in API level 26, commenting out for now
-
-            case "STREAM_ACCESSIBILITY":
-                audioStreamType = AudioManager.STREAM_ACCESSIBILITY;
-                break;
-            */
-            case "STREAM_ALARM":
-                audioStreamType = AudioManager.STREAM_ALARM;
-                break;
-            case "STREAM_DTMF":
-                audioStreamType = AudioManager.STREAM_DTMF;
-                break;
-            case "STREAM_MUSIC":
-                audioStreamType = AudioManager.STREAM_MUSIC;
-                break;
-            case "STREAM_NOTIFICATION":
-                audioStreamType = AudioManager.STREAM_NOTIFICATION;
-                break;
-            case "STREAM_RING":
-                audioStreamType = AudioManager.STREAM_RING;
-                break;
-            case "STREAM_SYSTEM":
-                audioStreamType = AudioManager.STREAM_SYSTEM;
-                break;
-            case "STREAM_VOICE_CALL":
-                audioStreamType = AudioManager.STREAM_VOICE_CALL;
-                break;
-            default:
-                audioStreamType = AudioManager.USE_DEFAULT_STREAM_TYPE;
-        }
+    private int speak(String utterance, String utteranceId, ReadableMap options) {
+        forcePhoneSpeaker = options.hasKey("KEY_OPTION_FORCE_PHONE_SPEAKER") ? options.getBoolean("KEY_OPTION_FORCE_PHONE_SPEAKER") : false;
+        isCarAudioSystem = options.hasKey("KEY_OPTION_CAR_AUDIO_SYSTEM") ? options.getBoolean("KEY_OPTION_CAR_AUDIO_SYSTEM") : false;
+        float volume = options.hasKey("KEY_OPTION_VOLUME") ? (float)options.getDouble("KEY_OPTION_VOLUME") : 1.0f;
+        float pan = options.hasKey("KEY_OPTION_PAN") ? (float)options.getDouble("KEY_OPTION_PAN") : 0.0f;
```

Comment:
```suggestion
        float volume = options.hasKey("KEY_OPTION_VOLUME") ? (float) options.getDouble("KEY_OPTION_VOLUME") : 1.0f;
        float pan = options.hasKey("KEY_OPTION_PAN") ? (float) options.getDouble("KEY_OPTION_PAN") : 0.0f;
```
Not a dealbreaker, but the original code was like this